doc/getting-started/generating-transactions.md

```diff
@@ -2,19 +2,19 @@
 
 ## TransactionBuilder API
 
-In order to simplify transaction creation, we provide a `TransactionBuilder` struct that manages witnesses, fee calculation, change addresses and such. Assume we have instantiated an instance under the variable `builder` for this explanation.
+In order to simplify transaction creation, we provide a `TransactionBuilder` struct that manages witnesses, fee calculation, change addresses and such. Assume we have instantiated an instance under the variable `builder` for this explanation. The `TransactionBuilder` requires several protocol parameters governing Cardano to be created which is shown in the following section. These are specified initially in the genesis file for Cardano nodes.
 
 The minimum required for a valid transaction is to add inputs, outputs, time-to-live and either set the fee explicitly with `builder.set_fee(fee)`, or calculate it implicitly using `builder.add_change_if_needed(address)`.
-Optionally a transaction can also have certificates, withdrawals, and metadata added to it.
+Optionally a transaction can also have certificates, reward withdrawals, and metadata added to it.
 Any change made to the builder can impact the size and thus the fee so the fee should be the last thing set.
-If implicitly setting the fee any extra ADA (`inputs + withdrawals - outputs - min fee`) is sent to the provided change address.
-Fees must be balanced, i.e. `inputs + withdrawals = outputs + fee` which must be manually ensured if you explicitly set the fee.
+If implicitly setting the fee any extra ADA (`inputs + withdrawals - outputs + refund - deposit - min fee`) is sent to the provided change address.
+Fees must be balanced, i.e. `inputs + withdrawals + refund = outputs + fee + deposit` which must be manually ensured if you explicitly set the fee.
```

Comment:
Fees don't have to be balanced. You can specify a higher fee than is required, and the extra fee gets burned.